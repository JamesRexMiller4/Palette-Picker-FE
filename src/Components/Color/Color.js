import React, { useState } from 'react';
import './Color.scss'

const Color = () => {
  return (
    <h2>it's color!</h2>
  )
}

export default Color;