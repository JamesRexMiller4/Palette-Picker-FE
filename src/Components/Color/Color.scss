.color-column {
  width: 20vw;
  height: fit-content;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
  // border-radius: 25px 25px;
}

.lock-icon {
  width: 5vw;
  height: 5vh;
    &:hover {
      cursor: pointer;
    }
}

.hidden {
  visibility: hidden;
}