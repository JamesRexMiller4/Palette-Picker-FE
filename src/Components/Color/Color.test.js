import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import Color from './Color.js';
import {shallow} from 'enzyme';
 
describe('Color Component', () => {
  let wrapper;
 
  beforeEach(() => {
    wrapper = shallow(<Color 
    color={'#FFFFFF'}
    locked={true}
    id={1}
    review={false}
    border={''}
    />)
  })
 
  it('it should match the snapshot with all the data passed through, locked set to true', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('it should match the snapshot with all the data passed through, locked set to false', () => {
    wrapper = shallow(<Color 
    color={'#FFFFFF'}
    locked={false}
    id={1}
    review={false}
    border={''}
    />)

    expect(wrapper).toMatchSnapshot();
  });
  it('it should match the snapshot with all the data passed through, review and locked set to true', () => {
    wrapper = shallow(<Color 
    color={'#FFFFFF'}
    locked={true}
    id={1}
    review={true}
    border={'25px 0 0 25px'}
    />)
    expect(wrapper).toMatchSnapshot();
  });
  it('it should match the snapshot with all the data passed through, review set to false and locked set to true', () => {
    wrapper = shallow(<Color 
    color={'#FFFFFF'}
    locked={false}
    id={1}
    review={true}
    border={'25px 0 0 25px'}
    />)
    expect(wrapper).toMatchSnapshot();
  });
  test.skip('dispatch should be called on icon click, with an action object', () => {
    const id = 1;
    render(wrapper)
    expect(screen.getByAltText(/an icon of a locked lock/i).src).toEqual("http://localhost/lock.svg")
    fireEvent.click(screen.getByAltText(/an icon of a locked lock/i))
    render(wrapper)
    expect(screen.getByAltText(/an icon of an unlocked lock/i).src).toEqual("http://localhost/unlock.svg")
  });
});