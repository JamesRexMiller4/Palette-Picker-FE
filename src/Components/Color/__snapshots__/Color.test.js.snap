// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component it should match the snapshot with all the data passed through, locked set to false 1`] = `
<Fragment>
  <div
    className="color-column"
    style={
      Object {
        "backgroundColor": "##FFFFFF",
      }
    }
  >
    <img
      alt="an icon of an unlocked lock"
      className="lock-icon"
      onClick={[Function]}
      src="unlock.svg"
    />
    <h2
      className="color-h2"
    >
      #
      #FFFFFF
    </h2>
  </div>
</Fragment>
`;

exports[`Color Component it should match the snapshot with all the data passed through, locked set to true 1`] = `
<Fragment>
  <div
    className="color-column"
    style={
      Object {
        "backgroundColor": "##FFFFFF",
      }
    }
  >
    <img
      alt="an icon of a locked lock"
      className="lock-icon"
      onClick={[Function]}
      src="lock.svg"
    />
    <h2
      className="color-h2"
    >
      #
      #FFFFFF
    </h2>
  </div>
</Fragment>
`;

exports[`Color Component it should match the snapshot with all the data passed through, review and locked set to true 1`] = `
<Fragment>
  <div
    className="color-column bubble"
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": "25px 0 0 25px",
      }
    }
  >
    <img
      alt="an icon of a locked lock"
      className="lock-icon hidden"
      onClick={[Function]}
      src="lock.svg"
    />
    <h2
      className="color-h2 hidden"
    >
      #FFFFFF
    </h2>
  </div>
</Fragment>
`;

exports[`Color Component it should match the snapshot with all the data passed through, review set to false and locked set to true 1`] = `
<Fragment>
  <div
    className="color-column bubble"
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": "25px 0 0 25px",
      }
    }
  >
    <img
      alt="an icon of an unlocked lock"
      className="lock-icon hidden"
      onClick={[Function]}
      src="unlock.svg"
    />
    <h2
      className="color-h2 hidden"
    >
      #FFFFFF
    </h2>
  </div>
</Fragment>
`;
